platform/security/sts/security-sts-pkivalidator/src/main/java/org/codice/ddf/security/validator/pki/PKITokenValidator.java

```diff
@@ -204,10 +209,38 @@ public TokenValidatorResponse validateToken(TokenValidatorParameters tokenParame
       Credential returnedCredential = validator.validate(credential, requestData);
       X500Principal subjectX500Principal =
           returnedCredential.getCertificates()[0].getSubjectX500Principal();
+
       response.setPrincipal(subjectX500Principal);
       if (response.getAdditionalProperties() == null) {
         response.setAdditionalProperties(new HashMap<>());
       }
+
+      try {
+        Collection<List<?>> subjectAlternativeNames =
+            returnedCredential.getCertificates()[0].getSubjectAlternativeNames();
+        if (CollectionUtils.isNotEmpty(subjectAlternativeNames)) {
+          for (List<?> alternativeNameData : subjectAlternativeNames) {
```

Comment:
Could we try and flatten this logic a bit with a stream and filters?

Reply:
Updated to use stream and separate method to make this a bit more readable.